dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ReferenceConfig.java

```diff
@@ -772,4 +774,24 @@ public Invoker<?> getInvoker() {
     public Runnable getDestroyRunner() {
         return this::destroy;
     }
+
+    @Override
+    public int getInvokerNum() {
+        Invoker<?> invoker = getInvoker();
+        if (invoker instanceof MigrationInvoker) {
+            AbstractDirectory<?> directory = (AbstractDirectory<?>) ((MigrationInvoker<?>) invoker).getDirectory();
+            return directory.getAllInvokers().size();
+        }
+        return super.getInvokerNum();
+    }
+
+    @Override
+    public int getValidInvokerNum() {
+        Invoker<?> invoker = getInvoker();
+        if (invoker instanceof MigrationInvoker) {
+            AbstractDirectory<?> directory = (AbstractDirectory<?>) ((MigrationInvoker<?>) invoker).getDirectory();
+            return directory.getValidInvokers().size();
+        }
+        return super.getValidInvokerNum();
+    }
 }
```

Comment:
`ReferenceConfig` itself should not depends on `MigrationInvoker`.
Get invokers size from consumer model.
See `org.apache.dubbo.qos.command.impl.Ls`.